generator/template_repo.go

```diff
@@ -207,6 +213,58 @@ func asPrettyJSON(data interface{}) (string, error) {
 	return string(b), nil
 }
 
+func goSliceInitializer(data interface{}) (string, error) {
+	// goSliceInitializer constructs a Go literal initializer from interface{} literals.
+	// e.g. []interface{}{"a", "b"} is transformed in {"a","b",}
+	// e.g. map[string]interface{}{ "a": "x", "b": "y"} is transformed in {"a":"x","b":"y",}
+	// NOTE: this is currently used to construct simple slice intializers for default values.
+	// This allows for nicer slice initializers for slices of primitive types and avoid systematic use for json.Unmarshal().
+	b, err := json.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+	return strings.Replace(strings.Replace(strings.Replace(string(b), "}", ",}", -1), "[", "{", -1), "]", ",}", -1), nil
+}
+
+func hasDefaultParam(parameters GenParameters) bool {
```

Comment:
it seems odd that this has to be done at render time. Why isn't this just set on the parameter groups etc?

Reply:
Was just easier to experiment with. I am replacing it with a struct member then.

Reply:
OK. Will do. Will add a HasSomeDefaults bool in the GenXXX structure for Params and Headers.
Will see later on if relevant to generalize this to models (there are some defaulting issues with models too... this story shall also be told)

Reply:
Ok tested on home-made unit test suite